cvat/apps/engine/utils.py

```diff
@@ -165,25 +165,22 @@ def define_dependent_job(
     if not should_be_dependent:
         return None
 
-    started_user_jobs = [
-        job
-        for job in queue.job_class.fetch_many(
-            queue.started_job_registry.get_job_ids(), queue.connection
-        )
-        if job and job.meta.get("user", {}).get("id") == user_id
-    ]
-    deferred_user_jobs = [
-        job
-        for job in queue.job_class.fetch_many(
-            queue.deferred_job_registry.get_job_ids(), queue.connection
-        )
+    keys = []
+    with queue.connection.pipeline() as pipeline:
         # Since there is no cleanup implementation in DeferredJobRegistry,
         # this registry can contain "outdated" jobs that weren't deleted from it
         # but were added to another registry. Probably such situations can occur
         # if there are active or deferred jobs when restarting the worker container.
-        if job and job.meta.get("user", {}).get("id") == user_id and job.is_deferred
+        for registry in [queue.started_job_registry, queue.deferred_job_registry]:
+            pipeline.zrange(registry.key, 0, -1)
+        pipeline.lrange(queue.key, 0, -1)
+        results = pipeline.execute()
+        keys = [bytes_key.decode('utf-8') for result in results for bytes_key in result]
+
+    all_user_jobs = [
+        job for job in queue.job_class.fetch_many(keys, queue.connection)
+        if job and job.meta.get("user", {}).get("id") == user_id
```

Comment:
How about checking that a job is not deferred?

Reply:
> we know

Hmm, I do not know such situations to be honest 
Do you have any specific scenarios to reproduce?

Reply:
Okay, now I saw that accidentally removed code you are talking about

Reply:
@Marishka17 

I returned this check